disperser/apiserver/server.go

```diff
@@ -282,7 +283,13 @@ func (s *DispersalServer) disperseBlob(ctx context.Context, blob *core.Blob, aut
 
 	s.logger.Debug("received a new blob dispersal request", "authenticatedAddress", authenticatedAddress, "origin", origin, "blobSizeBytes", blobSize, "securityParams", strings.Join(securityParamsStrings, ", "))
 
-	if s.ratelimiter != nil {
+	// If paymentHeader is not empty, we use the meterer, otherwise we use the ratelimiter if the ratelimiter is available
+	if *paymentHeader != (core.PaymentMetadata{}) {
```

Comment:
Need to check if paymentHeader is nil first

Reply:
interesting 🤔 the header passed in should either be empty by `DisperseBlob` or non-empty by `DispersePaidBlob`, so thought there won't be a nil case. Added anyway for safety